lib/puppet/provider/ecs_service/v2.rb

```diff
@@ -0,0 +1,106 @@
+require_relative '../../../puppet_x/puppetlabs/aws.rb'
+
+
+Puppet::Type.type(:ecs_service).provide(:v2, :parent => PuppetX::Puppetlabs::Aws) do
+  confine feature: :aws
+
+  mk_resource_methods
+
+  def self.instances
+    results = ecs_client.list_clusters().cluster_arns.collect do |cluster_arn|
+      Puppet.debug("Working on #{cluster_arn}")
+      service_arns = ecs_client.list_services({cluster: cluster_arn}).service_arns
+
+      unless service_arns.size > 0
+        Puppet.debug("Skipping #{cluster_arn} due to empty service list")
+        next
+      end
+
+      cluster_name = cluster_arn.match(/^(.*)\/([_\w\d-]+)$/)[2]
+
+      response = ecs_client.describe_services({
```

Comment:
I need to check for `next_token` here to ensure that we collect all available services.